plugins/module_utils/module_helper.py

```diff
@@ -133,10 +152,60 @@ def text(self):
         return self.msg or str(self.exc_val)
 
 
+class NamedDeprecation(object):
+    def __init__(self, deprecation):
+        self.name = None
+        self.deprecation = deprecation
+        self._prepared_deprecation = None
+        self.acked = False
+        self.triggered = False
+        self.module = None
+        self.ack_param = None
+
+    def prepare(self, name, module, ack_param):
+        self.name = name
+        self.module = module
+
+        d = list(self.deprecation)
+        d[0] = (
+            "{0}"
+            "\n\n"
+            "To prevent this message, add the deprecation name '{2}' to the '{1}' module parameter:\n"
+            "  {1}:\n"
+            "    - {2}"
+        ).format(d[0], ack_param, name)
+        self._prepared_deprecation = d
+
+    def acknowledge(self):
+        self.acked = True
+
+    def trigger(self):
+        if self.triggered or self.acked:
+            return
+
+        self.module.deprecate(*self._prepared_deprecation)  # pylint: disable=ansible-deprecated-no-version
+        self.triggered = True
+
+    def __str__(self):
+        return "<NamedDeprecation: {0} {1}{2}{3}>".format(
+            self.name,
+            self.deprecation,
+            ' ACK' if self.acked else "",
+            ' TRG' if self.triggered else "",
+        )
+
+
 class ModuleHelper(object):
     _dependencies = []
     module = {}
     facts_name = None
+    named_deprecations = {}
+    ack_deprecations_module_param = "ack_named_deprecations"
+
+    class PrepareNamedDeprecation(object):
+        @staticmethod
```

Comment:
Could you add a comment which explains why this helper is needed and why the function must be called `deprecate` and have the same signature as `AnsibleModule.deprecate`?